WordPress/src/test/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCaseTest.kt

```diff
@@ -0,0 +1,256 @@
+
+                // Assert
+                assertThat(result.size).isGreaterThanOrEqualTo(2)
+                assertThat((result[1] as PostFollowStatusChanged).following).isTrue()
+            }
+
+    @Test
+    fun `follow external feed post action returns info to not show enable notification`() =
+            testWithUnFollowedSitePost {
+                // Given
+                whenever(readerUtilsWrapper.isExternalFeed(anyLong(), anyLong())).thenReturn(true)
+
+                // Act
+                val result = useCase.toggleFollow(readerPost).toList(mutableListOf())
+
+                // Assert
+                assertThat((result[0] as PostFollowStatusChanged).showEnableNotification).isFalse()
+            }
+
+    @Test
+    fun `un-follow external feed post action returns info to not show enable notification`() =
+            testWithFollowedSitePost {
+                // Given
+                whenever(readerUtilsWrapper.isExternalFeed(anyLong(), anyLong())).thenReturn(true)
+
+                // Act
+                val result = useCase.toggleFollow(readerPost).toList(mutableListOf())
+
+                // Assert
+                assertThat((result[0] as PostFollowStatusChanged).showEnableNotification).isFalse()
+            }
+
+    private fun createDummyReaderPost(id: Long): ReaderPost = ReaderPost().apply {
+        this.postId = id
+        this.blogId = id
+        this.title = "DummyPost"
+    }
+
+    private fun <T> testWithFollowedSitePost(block: suspend CoroutineScope.() -> T) {
+        test {
+            readerPost.isFollowedByCurrentUser = true
+            whenever(readerPostTableWrapper.isPostFollowed(readerPost)).thenReturn(true)
+            whenever(readerBlogActionsWrapper.followBlogForPost(anyOrNull(), anyBoolean(), anyOrNull())).then {
+                (it.arguments[2] as ActionListener).onActionResult(true)
+                true
+            }
+
+            block()
+        }
+    }
+
+    private fun <T> testWithUnFollowedSitePost(block: suspend CoroutineScope.() -> T) {
+        test {
+            readerPost.isFollowedByCurrentUser = false
+            whenever(readerPostTableWrapper.isPostFollowed(readerPost)).thenReturn(false)
+            whenever(readerBlogActionsWrapper.followBlogForPost(anyOrNull(), anyBoolean(), anyOrNull())).then {
+                (it.arguments[2] as ActionListener).onActionResult(true)
+                true
+            }
+
+            block()
+        }
+    }
+
+    private fun <T> testWithFailedResponseForFollowedSitePost(block: suspend CoroutineScope.() -> T) {
+        test {
+            readerPost.isFollowedByCurrentUser = true
+            whenever(readerPostTableWrapper.isPostFollowed(readerPost)).thenReturn(true)
+            whenever(readerBlogActionsWrapper.followBlogForPost(anyOrNull(), anyBoolean(), anyOrNull())).then {
+                (it.arguments[2] as ActionListener).onActionResult(false)
+                false
```

Comment:
I believe the code returns true from `followBlogForPost` and invokes `onActionResult(false)` when a request fails. We are returning false even from `followBlogForPost` here - is it intentional?

Reply:
We do use the value returned from `followBlogForPost`, instead 
- first emit `PostFollowStatusChanged` for optimistic update [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt#L38)
- then use the return (boolean) value of the last suspension point [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt#L45) 

So it seems return value from `followBlogForPost` doesn’t matter in the refactored code. But I see your point and have made a change here: 3912425